### PR TITLE
refactor: add SfCheckbox to SfFilter 

### PR DESCRIPTION
**Pull request by aniamusial**
# Related issue
#591 

# Scope of work
Replaced underline behavior with checkbox and isActive prop

# Checklist

- [ ] I followed [composition rules](https://docs.storefrontui.io/contributing/coding-guidelines.html#component-rules) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/vue/src/components/molecules/SfFilter/SfFilter.stories.js [hunk 1]
```diff
@@ -8,81 +8,46 @@ storiesOf("Molecules|Filter", module)
...
-      selected: {
-        default: boolean("selected", true, "Props")
+      isActive: {
+        default: boolean("isActive", true, "Props")
       }
     },
     template: `<div style="max-width: 300px">
        <SfFilter
-        :color="color"
+        v-model="isActive"
```
Comment (filrak) [inline packages/vue/src/components/molecules/SfFilter/SfFilter.stories.js]: It's either `v-model` or prop, I'd go with `v-model`

File: packages/vue/src/components/molecules/SfFilter/SfFilter.vue [hunk 2]
```diff
@@ -30,14 +37,23 @@ export default {
...
-      type: String,
-      default: ""
-    },
     selected: {
       type: Boolean,
       default: false
     }
+  },
+  computed: {
+    style() {
```
Comment (pspaczek) [inline packages/vue/src/components/molecules/SfFilter/SfFilter.vue]: @aniamusial you should use css for that

File: packages/vue/src/components/molecules/SfFilter/SfFilter.vue [hunk 3]
```diff
@@ -1,26 +1,33 @@
...
+      />
     </slot>
-    <!-- @slot  -->
     <slot name="count" v-bind="{ count }">
       <div class="sf-filter__count">{{ count }}</div>
     </slot>
   </div>
 </template>
 <script>
+// @vue/component
```
Comment (pspaczek) [inline packages/vue/src/components/molecules/SfFilter/SfFilter.vue]:
```suggestion
```

File: packages/vue/src/components/molecules/SfFilter/SfFilter.vue [hunk 4]
```diff
@@ -1,26 +1,33 @@
...
     <!-- @slot  -->
     <slot name="label" v-bind="{ label }">
-      <div class="sf-filter__label">{{ label }}</div>
+      <SfCheckbox
+        class="sf-filter__checkbox"
+        name="label"
+        :label="label"
+        :style="style"
+        :selected="selected"
+        @input="setActive"
```
Comment (pspaczek) [inline packages/vue/src/components/molecules/SfFilter/SfFilter.vue]:
```suggestion
        @input="this.$emit("input", !selected);"
```

**Comment by filrak** [comment]
image

**Pull request merged**